clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -186,16 +186,18 @@ ClientRequest newClientRequest(String nodeId, AbstractRequest.Builder<?> request
      * @param createdTimeMs the time in milliseconds to use as the creation time of the request
      * @param expectResponse true iff we expect a response
      * @param requestTimeoutMs Upper bound time in milliseconds to await a response before disconnecting the socket and
-     *                         cancelling the request. The request may get cancelled sooner if the socket disconnects
-     *                         for any reason including if another pending request to the same node timed out first.
+*                         cancelling the request. The request may get cancelled sooner if the socket disconnects
+*                         for any reason including if another pending request to the same node timed out first.
      * @param callback the callback to invoke when we get a response
+     * @param initialPrincipalName the initial client principal name, when building a forward request
```

Comment:
It seems like we only want this to be set when forwarding a request to the controller, right?  So the JavaDoc should discuss the fact that this can be null (or whatever the "not set" value is)